javascript-client/tutorial.md

```diff
@@ -266,24 +266,46 @@ Also, our sample app works directly with Metamask using `windows.ethereum`
 
 1. First add GSN to your project:
 ```
-yarn add @opengsn/provider @opengsn/contracts @opengsn/dev
+yarn add @opengsn/provider @opengsn/contracts
+```
+
+Also you will need the following dev dependencies to run your project locally:
+```
+yarn add --dev @opengsn/dev @opengsn/cli
 ```
 
 2. When you test your project locally, you usually start a local node using `hardhat node`.
-   To start a node with GSN deployed (and a relayer running), you should run:
+In order to start a Hardhat node run:
+```
+npx hardhat node
+```
+
+_Note that using `hardhat-deploy` plugin with the GSN is currently not supported. The GSN requires the Hardhat to run in a
+standalone process and the deployment script requires the GSN contracts making a circular dependency.
+As a workaround you can have a separate Hardhat config file for `hardhat-deploy` in a subdirectory (see step #4)._
+
+3. In order to deploy GSN contracts and start a Relay Server run:
    ```
-   npx gsn start --withNode
+   npx gsn start
    ``` 
-   This command will start `hardhat node` in the background, and then install GSN contracts, and finally will start a local relayer service. 
+   This command will install GSN contracts and start a local Relay Server.
    It also deploys an "accept-everything" paymaster that can be used to pay for all transactions.
-   (note that you can also run `hardhat node` in one terminal, and `gsn start` in another)
 
-4. Now in the source code, we need to locate the access to `windows.ethereum`,  
+4. Deploy your contracts. If you are using `hardhat-deploy` plugin, do:
+   ```
+   cd deployment
```

Comment:
you don't need to "cd" into deployment. 
also, why would you give instructions on using this plugin? I never created a "deployment" folder - the plugin creates "deployments/xxx" folders for each network, and manager this "deployments" folder itself.

Reply:
Removed any direct mention of `hardhat-deploy` here. You are right, this is not part of the GSN-specific flow.